### PR TITLE
Microverse > Friendship v2 >Milestone 6

### PR DESCRIPTION
**Pull request by jurgen1c**
In this milestone we: 
- Add link for Add friend & Unfriend
- Add link for approve & decline
- Show pending & sent requests
- Show approved friends

### CODE REVIEW

**Review by trekab — COMMENTED** [review]
### Invalid! :stop_sign: 
Hi team, it is unfortunate that I have to mark your code review request as invalid because you didn't follow the proper [Git FLow](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59882084/) as required on this project.

__Kindly make the necessary changes and submit another code review request.__
_Feel free to reach me on slack for any inquiries about this review._ :wink: 
__Treasure Kabareebe__
[GitHub](https://github.com/trekab) | [LinkedIn](https://www.linkedin.com/in/treasure-kabareebe/) | [Twitter](https://twitter.com/TKabareebe)

File: app/views/users/show.html.erb
Comment (christianotieno) [inline app/views/users/show.html.erb]: ## Remove Logic from this view

- [x] Let's make sure that we have NO application logic in any of our Views. We can encapsulate our view related logic in View helpers (app/helpers/\*_helper.rb) [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright.

**Review by christianotieno — CHANGES_REQUESTED** [review]
# Changes required ♻️ 
Hey team 👋 
Please find some of my comments on the impending changes required before we move on with this milestone, down below 👇  thank you.

- [x] I can not run the migrations as you have not exposed your database credential from `config/database.yml`  file. As a result, it is hard to robustly test your code. FIx this, kindly.

### Happy coding 💻 

#### Christian Otieno (TSE)

**Comment by jurgen1c** [comment]
I believe we have resolved mentioned issues

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello Team
### You have done well so far :clap: However, you have a few issues to fix before the project can be approved.
- [ ] When I click the add friend button, it gives me an error, see below
friend
### Make the changes and submit for another code review. Good luck!
- Happy coding!

File: config/routes.rb
Comment (rOluochKe) [inline config/routes.rb]: ### This repetition, let pass these actions in the resources friendships

File: config/routes.rb
Comment (addod19) [inline config/routes.rb]: We are asked to address these routes from the previous TSE :+1: Let us avoid repeating controller actions

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @jurgen1c and @RaminMammadzada, great job with the project :smile: We are almost done with the project, just a few comments to address so we can move ahead

- Four of the test cases are failing, kindly check and fix them
testtt
- `config/routes.rb`, we were asked to pass these actions in resource friendships to avoid repetition. As I can tell now, it has not been addressed. Kindly address that. :+1: 

[Optional]
- `application.html.erb` has logic in it. Let us fix it like how we did to the users' show view

Once you are done with these, kindly submit another code review :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by christianotieno — APPROVED** [review]
# Project Approved 🍾 
Hello team 👋 
The project has met all the required changes needed for approval. 👍 
I will go ahead and mark it as complete. Congratulations on the hard work. 🎆  

I wish you all the very best in your next tasks 🥇 
### Happy coding 💻 

#### Christian Otieno (TSE)

**Pull request merged**
